### PR TITLE
feat(clipboard): add tooltip message for confirmation

### PR DESCRIPTION
**Pull request by skhamvon**
ODS-613
- Add 2 slot to handle confirmation message
- Add surface to show the confirmation message as tooltip

### CODE REVIEW

File: packages/components/clipboard/src/components/osds-clipboard/osds-clipboard.scss [hunk 1]
```diff
@@ -17,3 +29,12 @@
         --ods-input-icon-cursor: not-allowed;
     }
 }
+
+ocdk-surface {
+  padding: 16px;
+  width: max-content;
+  max-width: 300px;
+  background: var(--white) 0% 0% no-repeat padding-box;
+  background: #FFFFFF 0% 0% no-repeat padding-box;
```
Comment (aesteves60) [inline packages/components/clipboard/src/components/osds-clipboard/osds-clipboard.scss:38]: Remove one of the 2 background

File: packages/components/clipboard/src/components/osds-clipboard/core/controller.ts [hunk 2]
```diff
@@ -1,8 +1,50 @@
...
+      this.component.surfaceMessage = successMessage;
+      if (this.component.surface && this.component.surfaceMessage !== "") {
+        this.component.surface.opened = !this.component.surface.opened;
+      }
+    } catch (error) {
+      this.component.surfaceMessage = errorMessage;
+      if (this.component.surface  && this.component.surfaceMessage !== "") {
+        this.component.surface.opened = !this.component.surface.opened;
+      }
+      return Promise.reject(error);
```
Comment (aesteves60) [inline packages/components/clipboard/src/components/osds-clipboard/core/controller.ts:26]:
```suggestion
      return throw error;
```

Comment (aesteves60) [reply packages/components/clipboard/src/components/osds-clipboard/core/controller.ts:26]: Not sure if the return is need with a throw

File: packages/components/clipboard/src/components/osds-clipboard/core/controller.ts [hunk 3]
```diff
@@ -1,8 +1,50 @@
...
+    this.component = component
+  }
+
   async handlerClick(value: string): Promise<void> {
-    await writeOnClipboard(value);
+    const successMessage = this.component.el.querySelector('[slot=success-message]')?.innerHTML;
+    const errorMessage = this.component.el.querySelector('[slot=error-message]')?.innerHTML;
+
+    try {
+      writeOnClipboard(value);
```
Comment (dpellier) [inline packages/components/clipboard/src/components/osds-clipboard/core/controller.ts:16]: either use `await` here or use `.then` / `.catch` but here the promise is just iggnored

File: packages/components/clipboard/src/components/osds-clipboard/osds-clipboard.tsx [hunk 4]
```diff
@@ -25,9 +34,21 @@ export class OsdsClipboard implements OdsClipboardAttribute, OdsClipboardEvent {
...
+
   /** @see OdsClipboardEvents.odsClipboardCopied */
   @Event() odsClipboardCopied!: EventEmitter<string>;
 
+  @Listen('click', { target: 'window' })
+  checkForClickOutside(event: any) {
+    this.controller.checkForClickOutside(event);
+  }
+
+  @State() surfaceMessage: string | undefined = "";
```
Comment (aesteves60) [inline packages/components/clipboard/src/components/osds-clipboard/osds-clipboard.tsx:50]: On top with the others variable

**Review by dpellier — CHANGES_REQUESTED** [review]

**Review by aesteves60 — CHANGES_REQUESTED** [review]
We need to edit the two message in storybook. 
- When no success message we have a tooltip undefined
image

File: packages/components/clipboard/src/components/osds-clipboard/core/controller.ts (hunk 3 above)
Comment (skhamvon) [reply packages/components/clipboard/src/components/osds-clipboard/core/controller.ts]: "await" has been added

File: packages/components/clipboard/src/components/osds-clipboard/osds-clipboard.scss (hunk 1 above)
Comment (skhamvon) [reply packages/components/clipboard/src/components/osds-clipboard/osds-clipboard.scss]: Second line removed

File: packages/components/clipboard/src/components/osds-clipboard/core/controller.ts (hunk 2 above)
Comment (skhamvon) [reply packages/components/clipboard/src/components/osds-clipboard/core/controller.ts]: Fixed

File: packages/components/clipboard/src/components/osds-clipboard/osds-clipboard.tsx (hunk 4 above)
Comment (skhamvon) [reply packages/components/clipboard/src/components/osds-clipboard/osds-clipboard.tsx]: variable has been moved

File: packages/components/clipboard/src/components/osds-clipboard/core/controller.ts [hunk 5]
```diff
@@ -1,8 +1,50 @@
...
+  closeSurface(): void {
+    if (this.component.surface && this.component.surface.opened) {
+      this.component.surface.close();
+    }
+  }
+
+  checkForClickOutside(event: any): void {
+    if (this.component.el.contains(event.target) || this.component.surface === undefined || !this.component.surface.opened) {
+      return;
+    } else {
```
Comment (aesteves60) [inline packages/components/clipboard/src/components/osds-clipboard/core/controller.ts:45]: else not need

File: packages/components/clipboard/src/components/osds-clipboard/core/controller.ts [hunk 6]
```diff
@@ -1,8 +1,50 @@
...
+    }
+  }
+
+  closeSurface(): void {
+    if (this.component.surface && this.component.surface.opened) {
+      this.component.surface.close();
+    }
+  }
+
+  checkForClickOutside(event: any): void {
```
Comment (aesteves60) [inline packages/components/clipboard/src/components/osds-clipboard/core/controller.ts:42]: The event is a MouseEvent

**Review by aesteves60 — CHANGES_REQUESTED** [review]

File: packages/components/clipboard/src/components/osds-clipboard/core/controller.ts (hunk 5 above)
Comment (skhamvon) [reply packages/components/clipboard/src/components/osds-clipboard/core/controller.ts]: Deleted

File: packages/components/clipboard/src/components/osds-clipboard/core/controller.ts (hunk 6 above)
Comment (skhamvon) [reply packages/components/clipboard/src/components/osds-clipboard/core/controller.ts]: Fixed

**Review by aesteves60 — APPROVED** [review]

**Review by astagnol — APPROVED** [review]

**Pull request merged**